dbt_invoke/properties.py

```diff
@@ -268,104 +278,158 @@ def migrate(
+                definition['resource_index']  = next(
+                    (index for (index, x) in enumerate(current_property_yaml[definition['resource_type_plural']]) if x['name'] == definition['resource_name'])
+                    , None
                 )
-                _LOGGER.info(f'Created {data["path"]}')
-            except Exception as e:
-                _LOGGER.error(f'Failed to create {data["path"]}: {e}')
+                definition['resource_properties'] = current_property_yaml[definition['resource_type_plural']][definition['resource_index']]
+                # Try to create each new property file. For each success, save
+                # the index of the resource to remove from the migration
+                # property file.
+                try:
+                    property_file_dict = _get_property_header(
+                        definition['resource_name'],
+                        definition['resource_type'],
+                        definition['resource_properties'],
+                    )
+                    _utils.write_yaml(definition['resource_destination'], property_file_dict, mode='x')
+                    indices_to_remove[definition['resource_type_plural']].append(
+                        definition['resource_index']
+                    )
+                    _LOGGER.info(f"Created {definition['resource_destination']}")
+                except Exception as e:
+                    _LOGGER.error(f"Failed to create {definition['resource_destination']}: {e}")
         # Remove migrated resources from migration_property_file_dict
         removed_counter = 0
         for resource_type_plural, indices in indices_to_remove.items():
             for i in sorted(indices, reverse=True):
-                migration_property_file_dict[resource_type_plural].pop(i)
+                current_property_yaml[resource_type_plural].pop(i)
                 removed_counter += 1
         # Overwrite the migration property file with its remaining
         # post-migration contents.
-        _utils.write_yaml(location, migration_property_file_dict)
+        _utils.write_yaml(location, current_property_yaml)
         _LOGGER.info(
-            f'Removed {removed_counter} migrated resources from'
-            f' {str(location.resolve())}'
+            f'Removed {str(removed_counter)} migrated resources from {str(location)}'
         )
-    else:
-        _LOGGER.info(f'Nothing to migrate from {str(location.resolve())}')
+        # Remove resource types from the source yaml
+        for resource_type_plural, resources in current_property_yaml.items():
+            if resource_type_plural in _SUPPORTED_RESOURCE_TYPES.values() and len(resources) == 0:
+                current_property_yaml.pop(resource_type_plural)
+        # Check if there are any properties left in the source file.
+        # If there aren't then delete the file.
+        if not set(_SUPPORTED_RESOURCE_TYPES.values()).intersection(current_property_yaml.keys()):
```

Comment:
I think we may often want to avoid deleting the property files for a couple of reasons:
- There are unsupported resource types like sources and exposures that may exist in these property files
- There may be comments in the property files outside of the model definitions that we do not migrate

What do you think about only deleting the property files that contain no non-whitespace characters?

Reply:
Yep that is a good point. I will have to take a read on how to determine this is true and make the change.

Reply:
So I started looking into it and I don't see an officially supported way in `ruamel.yaml`'s documentation (https://yaml.readthedocs.io/en/latest/detail.html?highlight=comments#round-trip-including-comments) to retrieve the comments and formatting. 

What are your thoughts on the following process:
1. Use Path.read_text() to read the contents of the newly overwritten property file as a string
2. `strip()` the string
3. Delete the file if the result of `strip()` is an empty string

Reply:
```suggestion
        if Path(location).read_text().strip() == "":
```

Reply:
Seems simple enough. I agree there isn't a great looking way to achieve it based on the interpretation of the yaml. I don't see much of an issue with reading the file again at this end stage for performance. Most of the time it will be empty.